src/eddymotion/utils.py

```diff
@@ -0,0 +1,176 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2022 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+"""Utils to sort the DWI data volume indices"""
+
+from enum import Enum
+
+import numpy as np
+
+
+class SortingStrategy(Enum):
+    """
+    Enum class representing different sorting strategies.
+
+    Available sorting strategies:
+    - LINEAR: Sorts the items in a linear order.
+    - RANDOM: Sorts the items in a random order.
+    - BVALUE: Sorts the items based on their b-value.
+    - CENTRALSYM: Sorts the items based on their central symmetry.
+    """
+
+    LINEAR = "linear"
+    RANDOM = "random"
+    BVALUE = "bvalue"
+    CENTRALSYM = "centralsym"
+
+
+def sort_dwdata_indices(dwdata, strategy, seed=None):
+    """Sort the DWI data volume indices following the given strategy.
+
+    Parameters
+    ----------
+    dwdata : :obj:`~eddymotion.dmri.DWI`
+        DWI dataset, represented by this tool's internal type.
+    strategy : :obj:`~eddymotion.utils.SortingStrategy`
+        The sorting strategy to be used. Available options are:
+        - SortingStrategy.LINEAR: Sort the indices linearly.
+        - SortingStrategy.RANDOM: Sort the indices randomly.
+        - SortingStrategy.BVALUE: Sort the indices based on the last column of gradients in ascending order.
+        - SortingStrategy.CENTRALSYM: Sort the indices in a central symmetric manner.
+    seed : :obj:`int` or :obj:`bool`, optional
+        Seed the random number generator. If an integer, the value is used to
+        initialize the generator; if ``True``, the arbitrary value
+        of ``20210324`` is used to initialize it.
+
+    Returns
+    -------
+    index_order : :obj:`numpy.ndarray`
+        The sorted index order.
+    """
+    if strategy == SortingStrategy.LINEAR:
+        return linear_action(dwdata)
+    elif strategy == SortingStrategy.RANDOM:
+        return random_action(dwdata, seed)
+    elif strategy == SortingStrategy.BVALUE:
+        return bvalue_action(dwdata)
+    elif strategy == SortingStrategy.CENTRALSYM:
+        return centralsym_action(dwdata)
+    else:
+        raise ValueError("Invalid sorting strategy")
+
```

Comment:
Because of later commits my comment has gone away. As we discussed, let's leave this for a future PR.
```suggestion
```